### PR TITLE
Adding model package for registry models, both on deployment and model operations

### PR DESCRIPTION
**Pull request by banibrata**
# Description

Please add an informative description that covers that changes made by the pull request and link all relevant issues.

If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.

# All SDK Contribution checklist:
- [ ] **The pull request does not introduce [breaking changes]**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [ ] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md).**

## General Guidelines and Best Practices
- [ ] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md##building-and-testing)
- [ ] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: sdk/ml/azure-ai-ml/azure/ai/ml/operations/_model_operations.py [hunk 1]
```diff
@@ -699,33 +722,106 @@ def package(self, name: str, version: str, package_request: ModelPackage, **kwar
...
+        #         "inputs": [],
+        #         "tags": {
+        #             "ModelPackageGenerated": "azureml://registries/azureml/models/distilbert-base-uncased/versions/5",
+        #             "ModelConfiguration": "Download, ",
+        #             "InferencingServerType": "AzureMLOnline"
+        #         },
+        #         "properties": "",
+        #         "skuArchitectureType": ""
+        #         }
+        # import debugpy
```
Comment (nemanjarajic) [inline sdk/ml/azure-ai-ml/azure/ai/ml/operations/_model_operations.py:791]: Remove commented out section

File: sdk/ml/azure-ai-ml/azure/ai/ml/operations/_model_operations.py [hunk 2]
```diff
@@ -699,33 +722,106 @@ def package(self, name: str, version: str, package_request: ModelPackage, **kwar
...
-            body=package_request,
-            **self._scope_kwargs,
-        ).result()
+        if hasattr(package_out, "target_environment_version"):
+            environment_version = package_out.target_environment_version
+        else:
+            environment_version = package_out.additional_properties["targetEnvironmentVersion"]
+
+        # environment_name = "diabetes-online-mlflow"
+        # environment_version = "12"
```
Comment (nemanjarajic) [inline sdk/ml/azure-ai-ml/azure/ai/ml/operations/_model_operations.py:807]: Remove commented section

Comment (banibrata) [reply sdk/ml/azure-ai-ml/azure/ai/ml/operations/_model_operations.py]: Nice catch, did not realize the content is not in sync with the latest local :)

File: sdk/ml/azure-ai-ml/azure/ai/ml/_scope_dependent_operations.py [hunk 3]
```diff
@@ -50,11 +55,15 @@ def __init__(
...
         registry_name: Optional[str] = None,
+        workspace_id: Optional[str] = None,
+        workspace_location: Optional[str] = None,
     ):
         self._subscription_id = subscription_id
         self._resource_group_name = resource_group_name
         self._workspace_name = workspace_name
         self._registry_name = registry_name
+        self._workspace_id = workspace_id
+        self._workspace_location = workspace_location
```
[the remark this answers is not in the document]
Comment (banibrata) [reply sdk/ml/azure-ai-ml/azure/ai/ml/_scope_dependent_operations.py:66]: These are not supposed to be exposed to ML_client users, this is only for registry path use, so I did not put any getter or setter to make it wide open.

File: sdk/ml/azure-ai-ml/azure/ai/ml/operations/_model_operations.py [hunk 4]
```diff
@@ -569,7 +596,8 @@ def share(self, name, version, *, share_with_name, share_with_version, registry_
 
         #  Get workspace info to get workspace GUID
         workspace = self._service_client.workspaces.get(
-            resource_group_name=self._resource_group_name, workspace_name=self._workspace_name
+            resource_group_name=self._resource_group_name,
+            workspace_name=self._workspace_name,
         )
         workspace_guid = workspace.workspace_id
```
Comment (nemanjarajic) [inline sdk/ml/azure-ai-ml/azure/ai/ml/operations/_model_operations.py:602]: Would this potentially cause an issue? Workspace doesnt have a "workspace_id" member so should this be "_workspace_id" or should workspace class include "workspace_id" in its __init__ method

File: sdk/ml/azure-ai-ml/azure/ai/ml/entities/_workspace/workspace.py [hunk 5]
```diff
@@ -251,6 +252,7 @@ def _from_rest_object(cls, rest_obj: RestWorkspace) -> "Workspace":
             feature_store_settings=feature_store_settings,
             enable_data_isolation=rest_obj.enable_data_isolation,
             workspace_hub=rest_obj.hub_resource_id,
+            workspace_id=rest_obj.workspace_id,
```
Comment (nemanjarajic) [inline sdk/ml/azure-ai-ml/azure/ai/ml/entities/_workspace/workspace.py:255]: If we are expecting rest_obj to have a "workspace_id" in the from_rest_object method should workspace_id be passed in the to_rest_object method? Looks to me like workspace_id will just be set to None here.

File: sdk/ml/azure-ai-ml/azure/ai/ml/operations/_model_operations.py (hunk 4 above)
Comment (banibrata) [reply sdk/ml/azure-ai-ml/azure/ai/ml/operations/_model_operations.py:602]: Let's revisit it, this is not directly related to the scope of this PR, but happy to chat, also it seems per your investigation the share feature for model works....

File: sdk/ml/azure-ai-ml/azure/ai/ml/entities/_workspace/workspace.py (hunk 5 above)
Comment (banibrata) [reply sdk/ml/azure-ai-ml/azure/ai/ml/entities/_workspace/workspace.py:255]: Good question: I think this ID is set by the service during the creation, so we don't explicitly pass it.

**Review by nemanjarajic — APPROVED** [review]

**Pull request merged**